src/hypervisor.c

```diff
@@ -107,6 +108,43 @@ cc_oci_expand_net_device_cmdline(struct cc_oci_config *config, guint index) {
 	return g_strdup("");
 }
 
+/*
+ *
+ *
+ */
+#define DPDK_CMDLINE_OBJ "memory-backend-file,id=dpdkmem,size=2048M,mem-path=/dev/hugepages,share=on,prealloc=on"
```

Comment:
Since the VM memory and DPDK memory sizes need to match, I think the best solution here is to update `hypervisor.args.in` so that the two VM memory sizes are specified with variable like this:

```
-m
@VM_MEM@,slots=2,maxmem=@VM_MEM_MAX@
```

The runtime will then expand:

- `@VM_MEM@` *and* the `size=` memory size in `DPDK_CMDLINE_OBJ` to the same `2G` value.
- `@VM_MEM_MAX@` to `3G`.

But we should also add the `--vm-mem-size=` and `--vm-mem-size-max=` global options to the runtime to allow that behaviour to be changed (without rebuilding) using `cc-oci-runtime.sh.cfg`.

A couple of other points:

- This new behaviour should be documented in the top-level `README.rst`.
- The niggle here is validating the arguments to `--vm-mem-size=` and `--vm-mem-size-max=`: we either have to add code that checks for the fairly flexible specifications that qemu permits (`1024`, `5M`, `2G`, etc), or defer to qemu when the hypervisor is launched. Maybe a half-way house approach would be to check that the argument is purely numeric with the exception of the last byte which can be alphabetic before handing it over to the hypervisor.
- We should add a `g_debug()` call to log if the user has specified `--vm-mem-size=` or `--vm-max-mem-size=` (along with the specified values).

@grahamwhaley - how does this sound to you?